### PR TITLE
Fix A2Grad diverging to NaN from an inverted running-mean weight

### PR DESCRIPTION
**Pull request by winklemad**
### Problem

`A2Grad` keeps a running mean of gradients — `delta_k = grad - avg_grad` — updated incrementally:

```python
avg_grad.add_(grad - avg_grad, alpha=group['step'] + 1)
```

The weight for an incremental mean is `1 / step`, but this uses `step + 1` — the **reciprocal**. With a growing weight the "average" overshoots the gradient every step and grows without bound, so the optimizer NaNs within ~35 steps for any input:

```
|avg_grad| : step5 ~39  step10 ~1.1e6  step20 ~7.7e17  step30 ~8.4e31  step35 inf  step36 NaN
```

```python
import torch
from pytorch_optimizer import A2Grad
p = torch.nn.Parameter(torch.rand(4))
opt = A2Grad([p], lr=1e-2)
for _ in range(100):
    opt.zero_grad(); ((p - 1.0) ** 2).sum().backward(); opt.step()
print(p)  # NaN on main
```

The existing tests only run a handful of iterations, so they never reach the divergence.

### Fix

Use `alpha = 1 / step` so `avg_grad` is the true running mean (verified: it then matches the exact arithmetic mean of the gradients to float precision, and the optimizer converges). Added `test_a2grad_running_mean_does_not_diverge`; `ruff` clean, existing A2Grad tests pass.

### CODE REVIEW

**Review by kozistr — DISMISSED** [review]
thanks! could you resolve the conflicts here too?

**Comment by winklemad** [comment]
Rebased and resolved the conflict here too (kept both regression tests in the shared file). Thanks for the review!
